entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryConverter.java

```diff
@@ -336,20 +339,24 @@ private AttributeFilter convertToAttributeFilter(
     return builder;
   }
 
-  private String convertToAttributeKey(RequestContext requestContext, Expression expression) {
+  private Pair<String, Boolean> convertToAttributeKey(
```

Comment:
so aside from my feelings about bringing in a second util lib, pair is generally a bad tool because it drops all semantics and it makes it harder for the reader to understand.`attributeData.getRight()` doesn't tell me what this is going to do, I have to read backwards to where that pair was generated to see what it holds. Much cleaner is too create a util class like

```java
  @Value
  private static class ColumnInformation {
    String columnName;
    boolean multiValued;
  }
```

now `builder.setIsMultiValued(columnInformation.isMultiValued())` and `String edsColumnName = columnInformation.geColumnName()` would be more readable.

Regardless, this function is no longer returning just a key and should be renamed.

Reply:
yeah on the proto side the col name is already a top level thing, so not sure we can really change it cleanly.

Reply:
I started with adding class, but there is no lombok here, so have to add all functions. And it will be used here only, so thought of going with Pair. I agree with readability part will add this class.

Reply:
done